osf/migrations/0137_update_schema_data.py

```diff
@@ -0,0 +1,254 @@
+        rs.save()
+
+def unset_schema_configs(state, schema):
+    RegistrationSchema = state.get_model('osf', 'registrationschema')
+    RegistrationSchema.objects.update(config=dict())
+
+def _generate_ridie_formblocks(state, question, page, _order, schema_id):
+    """ Builds formblocks from question information. RIDIE schemas are flat, with the
+        exception of file-upload questions, which appear to be nested based on "type",
+        but actually should not be.
+
+    :param StateApps state: Current Django model state
+    :param dict question: Question to build from
+    :param dict page: Page for context
+    :param int _order: Order for the next form block
+    :ret tuple(list, int): Returns a tuple of (<List of built blocks>, <_order for next block>)
+    """
+    RegistrationFormBlock = state.get_model('osf', 'registrationformblock')
+    blocks = []
+    qid = question['qid']
+    _type = 'osf-upload' if question['type'] == 'object' else FORMAT_TYPE_TO_TYPE_MAP[(question['format'], question['type'])]
+    block = RegistrationFormBlock(
+        page=page['title'],
+        block_id=qid,
+        section=question['title'],
+        block_type=_type,
+        block_text=question['description'],
+        help_text=question.get('help', ''),
+        choices=question.get('options', []),
+        required=question.get('required', True),
+        schema_id=schema_id,
+        _order=_order,
+    )
+    blocks.append(block)
+    _order += 1
+    return blocks, _order
+
+def _generate_formblocks(state, question, page, _order, schema_id):
+    """ Builds formblocks from question information.
+
+    :param StateApps state: Current Django model state
+    :param dict question: Question to build from
+    :param dict page: Page for context
+    :param int _order: Order for the next form block
+    :ret tuple(list, int): Returns a tuple of (<List of built blocks>, <_order for next block>)
+    """
+    RegistrationFormBlock = state.get_model('osf', 'registrationformblock')
+    blocks = []
+    _type = question['type']
+    qid = question['qid']
+    if _type == 'object':
+        if question.get('description', False):
+            # Question header create
+            block = RegistrationFormBlock(
+                page=page['title'],
+                block_id=qid,
+                section=question['title'],
+                block_type='header',
+                block_text=question['description'],
+                help_text=question.get('help', ''),
+                required=False,
+                schema_id=schema_id,
+                _order=_order,
+            )
+            blocks.append(block)
+            _order += 1
+        for obj in question['properties']:
+            _qid = obj['id']
+            if obj['type'] == 'object':
+                if obj.get('description', False):
+                    # Question sub-header create
+                    block = RegistrationFormBlock(
```

Comment:
Kinda a nit pick, but this would be much cleaner if it was recursive. Or at least delegated FormBlock creation to a helper.

Reply:
I was planning on delegating to a helper, but got sidetracked by other tasks before implementing.